### PR TITLE
Clone and other basic API

### PR DESCRIPTION
**Pull request by Thundermaker300**
Note: Idk why github made a mess of the commits, but changes are correct
* Added Clone() methods to Badge, Broadcast, ArmorAmmoLimit, and AttachmentIdentifier
* Added TeslaGate.NetIdentity
* Added Primitive.GameObject, Primitive.Transform, Primitive.NetIdentity
* Added Light.GameObject, Light.Transform, Light.NetIdentity
* Added Player.CountItem(ItemCategory)
* Minor doc changes in cassie

### CODE REVIEW

**Comment by Thundermaker300** [comment]
git moment

File: Exiled.API/Features/Badge.cs [hunk 1]
```diff
@@ -52,5 +52,14 @@ public Badge(string text, string color, int type, bool isGlobal = false)
...
         /// <returns>A string containing Badge-related data.</returns>
         public override string ToString() => $"{Text} ({Color}) [{Type}] *{IsGlobal}*";
+
+        /// <summary>
+        /// Clones the current <see cref="Badge"/> object.
+        /// </summary>
+        /// <returns>A new <see cref="Badge"/> object.</returns>
+        public Badge Clone()
+        {
+            return new(Text, Color, Type, IsGlobal);
```
Comment (Ficus-x) [inline Exiled.API/Features/Badge.cs:62]: Why dont you use lambda?

Comment (Thundermaker300) [reply Exiled.API/Features/Badge.cs:62]: That is a good question! I may have forgot it existed. will fix later

File: Exiled.API/Features/Door.cs [hunk 2]
```diff
@@ -372,6 +372,7 @@ public static void UnlockAll(Func<Door, bool> predicate)
         /// <summary>
         /// Breaks the specified door. No effect if the door cannot be broken, or if it is already broken.
         /// </summary>
+        /// <param name="type">The <see cref="DoorDamageType"/> to apply to the door.</param>
         /// <returns><see langword="true"/> if the door was broken, <see langword="false"/> if it was unable to be broken, or was already broken before.</returns>
         [Obsolete("BreakDoor() will be obsolete in future versions, please use BreakDoor(DoorDamageType)", false)]
         public bool BreakDoor() => BreakDoor(DoorDamageType.ServerCommand);
```
Comment (IRacle1) [inline Exiled.API/Features/Door.cs:378]: Ew, why?

Comment (Thundermaker300) [reply Exiled.API/Features/Door.cs:378]: I didn't even make this change wtf, look at changed files

**Comment by Thundermaker300** [comment]
closing because broken moment

**Pull request closed without merging**